plugins/inputs/knx_listener/knx_listener_test.go

```diff
@@ -0,0 +1,256 @@
+package knx_listener
+
+import (
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf/testutil"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/vapourismo/knx-go/knx"
+	"github.com/vapourismo/knx-go/knx/cemi"
+	"github.com/vapourismo/knx-go/knx/dpt"
+)
+
+const epsilon = 1e-6
+
+// Check if we can receive data
+func TestRegularReceives_DPT1(t *testing.T) {
+	acc := &testutil.Accumulator{}
+
+	// Setup the unit-under-test
+	listener := KNXListener{
+		ServiceType:    "dummy",
+		ServiceAddress: "manual",
+		Measurements: []Measurement{
+			{"test_1001", "1.001", []string{"1/0/1"}},
+			{"test_1002", "1.002", []string{"1/0/2"}},
+			{"test_1003", "1.003", []string{"1/0/3"}},
+		},
+	}
+
+	err := listener.Start(acc)
+	require.NoError(t, err)
+
+	// Get the dummy interface to send data to the unit-under-test
+	client := listener.client.(*KNXDummyInterface)
+
+	tstart := time.Now()
+
+	// Send some test data we expect in the config
+	client.Send(knx.GroupEvent{
+		Command:     knx.GroupWrite,
+		Destination: cemi.NewGroupAddr3(1, 0, 1),
+		Data:        dpt.DPT_1001(true).Pack(),
+	})
+	client.Send(knx.GroupEvent{
+		Command:     knx.GroupWrite,
+		Destination: cemi.NewGroupAddr3(1, 0, 2),
+		Data:        dpt.DPT_1002(false).Pack(),
+	})
+	client.Send(knx.GroupEvent{
+		Command:     knx.GroupWrite,
+		Destination: cemi.NewGroupAddr3(1, 0, 3),
+		Data:        dpt.DPT_1003(true).Pack(),
+	})
+
+	// Give the accumulator some time to collect the data
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Would be a bit better if we only waited a couple ms, or looped until we get the metrics we're waiting for, a few ms at a time. We try to keep the tests from sleeping.

Reply:
Changed now to ten milliseconds. However, without this sleep I sometimes get only a subset of the measurements and thus the unit-test fails. I have no idea what a good value is... Is there a more reliable way to wait for the accumulator to finish up?

Reply:
You could wait 2 ms and check to see what the count is in a loop until you have all the items you expect. it's not perfect (you could have timeouts), but it would be fast in the happy path. There's often a way to do it without sleeping, but it might require bigger changes.

Reply:
How about?
```
	// Give the accumulator some time to collect the data
	time.Sleep(2 * time.Millisecond)
	if len(acc.Metrics) != 3 {
		time.Sleep(10 * time.Millisecond)
	}
```
or even
```
	// Give the accumulator some time to collect the data
	if len(acc.Metrics) != 3 {
		time.Sleep(10 * time.Millisecond)
	}
```